test/lit/passes/remove-unused-brs-gc.wast

```diff
@@ -494,8 +494,10 @@
  ;; CHECK-NEXT:   )
  ;; CHECK-NEXT:   (drop
  ;; CHECK-NEXT:    (br $block
- ;; CHECK-NEXT:     (local.tee $any
- ;; CHECK-NEXT:      (struct.new_default $struct2)
+ ;; CHECK-NEXT:     (ref.as_non_null
```

Comment:
TypeGeneralizing would be able to then optimize out this extra cast. It would be great to find time to finish that :)